### PR TITLE
nodePackages.vim-language-server: init at 2.1.0

### PR DESCRIPTION
**Pull request by leungbk**
###### Motivation for this change
Add vim-language-server, a LSP-compliant server for Vimscript

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by matt-snider — COMMENTED** [review]
@leungbk Nice addition! I was just looking to use this myself and noticed it was missing.

The changes (besides conflicts) look good to me. I built it from your branch and tried it out in vim with [LanguageClient-neovim](https://github.com/autozimu/LanguageClient-neovim) - `let g:LanguageClient_serverCommands.vim = ['/nix/store/.../vim-language-server', '--stdio']` - and it works :-)

@calbrecht I haven't reviewed any PRs yet in nixpkgs, but I'd like to start. Since you are the suggested reviewer, is there anything else to be checked usually? I did read [the relevant documentation](https://nixos.org/nixpkgs/manual/#chap-reviewing-contributions), but I'm not clear on how one should review changes in `node-packages.nix` or similar, to ensure that the appropriate generated changes are present and haven't been tinkered with or done erroneously.

**Review by matt-snider — APPROVED** [review]

**Comment by calbrecht** [comment]
@matt-snider although beeing suggested as reviewer, i did not do so much reviews already. But i know there is this [chapter about reviewing](https://nixos.org/nixpkgs/manual/#chap-reviewing-contributions) in nixpkgs manual.

**Comment by matt-snider** [comment]
@calbrecht Thanks for the response. I think the issue I mentioned with `node-packages.nix` is probably irrelevant since any incorrect changes will be overwritten in later revisions anyways. Doesn't seem important.

@leungbk It looks good to me, and like I said, it works, but I don't have merge rights for this repo, so for now I can just review and wait for someone else to merge it

**Pull request merged**
